themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -2,153 +2,176 @@
+            return;
+        }
+        // display the error message on each of the ajax status place holder
+        $('.js-item-pending').addClass('text-danger').append(response.responseJSON.data);
+    },
+    itemQueueAjax: function(id, el){
+        clearTimeout(this.itemStatusTimer);
+        this.itemStatusIds.push(id);
+        this.itemStatusEls[id] = el;
+        this.itemStatusTimer = setTimeout(this.runItemAjaxForQueue.bind(this), this.itemStatusDelay);
+        el.addClass('js-item-pending').removeClass('hidden');
+        el.find('.status').removeClass('hidden');
+    },
+
+    runItemAjaxForQueue : function(){
+        if (this.itemStatusRunning) {
+            this.itemStatusTimer = setTimeout(this.runItemAjaxForQueue.bind(this), this.itemStatusDelay);
+            return;
+        }
+        $.ajax({
+            dataType: this.dataType,
+            method: this.method,
+            url: VuFind.path+this.url,
+            context: this,
+            data: { 'id': this.itemStatusIds }
+        })
+            .done(this.checkItemStatusDone)
+            .fail( this.itemStatusFail)
+            .always(function(){
+                this.itemStatusRunning = false;
+            });
+    }//end runItemAjax
+};
+
+//store the handlers in a "hash" obj
+//add you own overridden handler here
+var handlers = {
+    'ils':ItemStatusHandler,
+    };
 
 function checkItemStatus(el) {
-  var $item = $(el);
-  if ($item.find('.hiddenId').length === 0) {
-    return false;
-  }
-  var id = $item.find('.hiddenId').val();
-  itemQueueAjax(id + '', $item);
+    var $item = $(el);
+    var handlerName='ils';
+    if ($item.hasClass('js-item-pending')) {
+        return;
+    }
+    if ($item.find('.hiddenId').length === 0) {
+        return false;
+    }
+    var id = $item.find('.hiddenId').val();
+    if ($item.find('.hiddenHandler').length > 0) {
+        handlerName=$item.find('.hiddenHandler').val();
+    }
+    if($item.data("handlerName")) {
+        handlerName=$item.data("handlerName");
+    }
+    //queue the element into the handler
+    handlers[handlerName].itemQueueAjax(id, $item);
 }
 
-var itemStatusObserver = null;
 function checkItemStatuses(_container) {
-  var container = _container instanceof Element
-    ? _container
-    : document.body;
+    var container = _container instanceof Element
+        ? _container
+        : document.body;
 
-  var ajaxItems = $(container).find('.ajaxItem');
-  for (var i = 0; i < ajaxItems.length; i++) {
-    var id = $(ajaxItems[i]).find('.hiddenId').val();
-    itemQueueAjax(id, $(ajaxItems[i]));
-  }
-  // Stop looking for a scroll loader
-  if (itemStatusObserver) {
-    itemStatusObserver.disconnect();
-  }
+    var ajaxItems = $(container).find('.ajaxItem');
+    for (var i = 0; i < ajaxItems.length; i++) {
+        var id = $(ajaxItems[i]).find('.hiddenId').val();
+        var handlerName = $(ajaxItems[i]).find('.handler').val();
```

Comment:
Should we have `.handler` here be the same class as `.hiddenHandler` on new line 142 above?

Reply:
I think it makes sense to make them the same.